### ISSUE TITLE
Support other strategies

### ISSUE THREAD

**Issue by DarkLite1** [jkettmann/graphql-passport#39]
Does `graphql-passport` also supports other strategies like say [passport-azure-ad](https://github.com/AzureAD/passport-azure-ad) with its `BearerStrategy`?`

The documentation only mentions `GraphQLLocalStrategy `, that's why it's unclear to me.

**Comment by gforge** [comment]
This should be doable but I've never worked with the more advanced strategies. The `GraphQLLocalStrategy` is implemented but it would be awesome if all of the strategies passport provides could be available.

**Comment by DarkLite1** [comment]
Indeed, that would really be cool! So users can just use the standard available passport strategies and you don't need to replicate them and in turn avoid code duplication. 

For the moment I'm following [your guide](https://jkettmann.com/authentication-and-authorization-with-graphql-and-passport/) to try and implement the `passport-azure-ad` strategie. The plan is to manually add the passport `login`  and `authenticate` methods to the `context` just like your `logout` example in the guide.

I'm a bit of a newbee to web development, so fingers crossed :)

**Comment by PacomeRivier** [comment]
Hello, I'm trying to do the exact same thing, any update on how you managed to implement it @DarkLite1 ?

**Comment by DarkLite1** [comment]
I couldn't wait for `graphql-passport` to support this feature so I used the standard `passport` to set up authentication with Azure AD.

**Comment by jkettmann** [comment]
Hey @DarkLite1, @PacomeRivier. Sorry for keeping you wait. The biggest pain that `graphql-passport` solves is local authentication with email and password. That's what the `GraphQLLocalStrategy` is for. Other strategies can be used in the normal way with `passport`.

If you're looking for an example you can see [this blog post](https://jkettmann.com/authentication-with-graphql-and-passport-js-the-frontend). [Here is the final repository](https://github.com/jkettmann/frontend-for-authentication-with-graphql-and-passport/blob/final-code/api/index.js) that combines the Facebook Strategy with local email/password.

If you don't use local email/password to authenticate the only thing you might use from this package is the `buildContext` function as you can see [here](https://github.com/jkettmann/frontend-for-authentication-with-graphql-and-passport/blob/final-code/api/index.js#L95). This basically just copies some of the passport functions like `authenticate`, `login`, `logout` and so on into the GraphQL context. So you might not necessarily need to use this package if you don't use local email/password.
